tools/cherry-pick/bin/run.js

```diff
@@ -0,0 +1,580 @@
+/**
+ * Tool to automate steps to cherry pick fixes into release branch.
+ *
+ * @package
+ */
+
+import fetch from 'node-fetch';
+import os from 'os';
+import fs from 'node:fs';
+import path from 'node:path';
+import { spawnSync, spawn } from 'node:child_process';
+import ora from 'ora';
+
+const args                  = process.argv.slice( 2 );
+const usage                 = 'Usage: pnpm cherry-pick <release_branch> <pull_request_number>. Separate pull request numbers with a comma (no space) if more than one.';
+const releaseBranch         = args[ 0 ];
+const prs                   = args[ 1 ];
+let   githubToken           = '';
+let   tempWooDir            = '';
+let   changelogsToBeDeleted = [];
+let   prCommits             = {};
+let   githubRemoteURL       = 'https';
+
+if ( typeof process.env.GITHUB_CHERRY_PICK_TOKEN === 'undefined' ) {
+	console.error( 'A GitHub token needs to be assigned to the "GITHUB_CHERRY_PICK_TOKEN" environment variable' );
+	process.exit( 1 );
+}
+
+if ( typeof process.env.GITHUB_REMOTE_URL !== 'undefined' ) {
+	githubRemoteURL = process.env.GITHUB_REMOTE_URL;
+}
+
+// If no arguments are passed.
+if ( ! args.length ) {
+	console.error( usage );
+	process.exit( 1 );
+}
+
+// If missing one of the arguments.
+if ( typeof releaseBranch === 'undefined' || typeof prs === 'undefined' ) {
+	console.error( usage );
+	process.exit( 1 );
+}
+
+// Accounts for multiple PRs.
+const prsArr           = prs.split( ',' );
+const version          = releaseBranch.replace( 'release/', '' );
+const cherryPickBranch = 'cherry-pick-' + version + '/' + prsArr.toString().replace( ',', '-' );
+githubToken            = process.env.GITHUB_CHERRY_PICK_TOKEN;
+
+async function getCommitFromPrs( prsArr ) {
+	let properties = {
+			method: 'GET',
+			headers: {
+				'Accept': 'application/vnd.github.v3+json',
+				'Authorization': `token ${githubToken}`
+			}
+		};
+
+	for ( const pr of prsArr ) {
+		const response = await fetch( 'https://api.github.com/repos/woocommerce/woocommerce/pulls/' + pr, properties );
+
+		if ( response.status !== 200 ) {
+			throw 'One or more of the PR reference was not found.';
+		}
+
+		await response.json().then( data => {
+			prCommits[ pr ] = data.merge_commit_sha;
+		} );
+	}
+}
+
+/**
+ * Checks out a branch.
+ *
+ * @param string branch The branch to checkout.
+ * @param boolean newBranch A flag to create a new branch.
+ */
+function checkoutBranch( branch, newBranch = false ) {
+	const spinner = ora( {
+		spinner: 'bouncingBar',
+		color: 'green'
+	} );
+
+	let output = [];
+
+	return new Promise( ( resolve, reject ) => {
+		let response = spawn( 'git', [ 'checkout', branch ] );
+
+		if ( newBranch ) {
+			response = spawn( 'git', [ 'checkout', '-b', branch ] );
```

Comment:
How does this play with the above `spawn` call? Technically we're starting two processes here, but, we're only listening to one, right?

Reply:
Yeah I can more it more explicit so it is only calling one or the other.